genesis/process/sovereignGenesisBlockCreator.go

```diff
@@ -277,3 +317,53 @@ func setSovereignStakedData(
 
 	return stakingTxs, nil
 }
+
+func createSovereignGenesisESDTTransfers(args ArgsGenesisBlockCreator, scrProcessor process.SmartContractResultProcessor) (map[string]data.TransactionHandler, error) {
+	initialESDTTxs := make(map[string]data.TransactionHandler, 0)
+	initialAccounts := args.AccountsParser.InitialAccounts()
+
+	tokenID := args.Config.SovereignConfig.GenesisConfig.NativeESDT
+	for nonce, initialAcc := range initialAccounts {
+		accInitialBalance := initialAcc.GetBalanceValue()
+
+		log.Debug("creating genesis initial esdt balance",
+			"address", initialAcc.GetAddress(), "balance", accInitialBalance.String(), "tokenID", tokenID)
+
+		scr := &smartContractResult.SmartContractResult{
+			Nonce:   uint64(nonce),
+			RcvAddr: initialAcc.AddressBytes(),
+			SndAddr: core.ESDTSCAddress,
+			Data:    createGenesisSCRData(tokenID, accInitialBalance),
```

Comment:
Why are you adding the same amount of `WEGLD` as of `EGLD` in the account?

Reply:
EGLD balance will "dissapear" in the future(will be replaced with esdts), we will have no "native balance". So instead of distributing egld at genesis, we will distribute esdt tokens